### PR TITLE
Get placement policies from contract

### PR DESCRIPTION
**Pull request by smallhive**

### CODE REVIEW

File: cmd/s3-gw/app.go [hunk 1]
```diff
@@ -190,6 +192,34 @@ func (a *App) initLayer(ctx context.Context, anonSigner user.Signer, neoFS *neof
 	}
 }
 
+func (a *App) initStoragePolicyProvider(ctx context.Context, contractName string, cfg *viper.Viper) *storagePolicyProvider {
+	w, err := neogoWallet.NewWalletFromFile(cfg.GetString(cfgRPCContractsStoragePolicyWallet))
```
[the remark this answers is not in the document]
Comment (smallhive) [reply cmd/s3-gw/app.go]: It was a question. It looks easier without a wallet, updated

File: cmd/s3-gw/storage_policy.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		// contract hash is not resolved.
+		if contractHash.Equals(zero) {
+			contractHash, err = resolveContract(cl, inv, contractName)
+
+			if err != nil {
+				return nil, fmt.Errorf("resolve contract: %w", err)
+			}
+		}
+
+		act, err := actor.NewSimple(cl, acc)
```
[the remark this answers is not in the document]
Comment (smallhive) [reply cmd/s3-gw/storage_policy.go]: Done

File: api/handler/api.go [hunk 3]
```diff
@@ -36,6 +38,13 @@ type (
 		Default() netmap.PlacementPolicy
 		Get(string) (netmap.PlacementPolicy, bool)
 	}
+
+	// PlacementPolicyProvider takes placement policy from contract.
+	PlacementPolicyProvider interface {
+		// GetPlacementPolicy get policy by name.
+		// Returns [models.ErrNotFound] if policy not found.
+		GetPlacementPolicy(name string) (*netmap.PlacementPolicy, error)
```
[the remark this answers is not in the document]
Comment (smallhive) [reply api/handler/api.go]: Updated

File: docs/configuration.md [hunk 4]
```diff
@@ -255,13 +255,15 @@ peers:
...
 | `region_mapping` | `string` | yes           |               | Path to file that maps aws `LocationConstraint` values to NeoFS placement policy. The similar to `--container-policy` flag in `neofs-s3-authmate` util, see in [docs](./authmate.md#containers-policy)                                       |
 | `locations`      | `map`    | yes           |               | Hashtable that maps aws LocationConstraint values to NeoFS placement policy. It's similar to the region_mapping config option, but allows for in-place configuration and extends/overrides values from region_mapping file if it's provided. |
+| `contract_name`  | `string` | yes           |               | The contract name to resolve policy name. An empty value disables feature. The contract must have method `ResolvePolicy(userAddr util.Uint160, policyName string)`.                                                                          |
```
Comment (roman-khimov) [inline docs/configuration.md:266]: Return value? And it's `resolvePolicy`, and types better be NEP-14.

Comment (smallhive) [reply docs/configuration.md]: Updated

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**